topology/probes/istio/virtualservice.go

```diff
@@ -71,6 +72,29 @@ func (vsSpec virtualServiceSpec) getAppsVersions() map[string][]string {
 	return appsVersions
 }
 
+func virtualServicePodMetadata(a, b interface{}, typeA, typeB, manager string) graph.Metadata {
+	vs := a.(*kiali.VirtualService)
+	pod := b.(*v1.Pod)
+	vsSpec := &virtualServiceSpec{}
+	if err := mapstructure.Decode(vs.Spec, vsSpec); err != nil {
+		return nil
+	}
+	m := k8s.NewEdgeMetadata(manager, typeA)
+	for _, http := range vsSpec.HTTP {
+		for _, route := range http.Route {
+			app := route.Destination.App
+			version := route.Destination.Version
+			if app == pod.Labels["app"] && version == pod.Labels["version"] {
+				if route.Weight > 0 {
+					m["weight"] = route.Weight
```

Comment:
Replace the entire if block lines 88-90 with:

```
m["weight"] += route.Weight
```

supporting:

```
    route:
    - destination:
        port:
          number: 81
        host: service3.default.svc.cluster.local
        subset: v1
      weight: 80
    - destination:
        port:
          number: 82
        host: service3.default.svc.cluster.local
        subset: v1
      weight: 20
```

while in this case should be `100` as we don't yet support `port` granularity

Reply:
DONE. 
Note that in some cases we don't want "weight" field to appear in the metadata, so the implementation is slightly different.